### PR TITLE
[DS-3714] REST Collection Report - Need a paginated findByCollection call that can return withdrawn items

### PR DESCRIPTION
**Pull request by terrywbrady**
See https://jira.duraspace.org/browse/DS-3714

The following tutorial page will provide instructions on how to replicate this error.
https://github.com/terrywbrady/restReportTutorial/blob/master/demo2/README.md

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/content/service/ItemService.java [hunk 1]
```diff
@@ -147,6 +147,18 @@
     public Iterator<Item> findAllByCollection(Context context, Collection collection) throws SQLException;
 
     /**
+     * Get all the items in this collection. The order is indeterminate.
+     *
```
Comment (pnbecker) [inline dspace-api/src/main/java/org/dspace/content/service/ItemService.java]: Could you please add a note how this is different from findByCollection? I assume one contains private and withdrawn items, the other doesn't?

**Review by pnbecker — COMMENTED** [review]
Beside my request to extend the javadoc a little bit, the code looks fine to me. I haven't tested it yet.
One thing that is not clear for me: is there still a way to easily see all archived items per community/collection in the rest report tool?

File: dspace-api/src/main/java/org/dspace/content/service/ItemService.java (hunk 1 above)
Comment (terrywbrady) [reply dspace-api/src/main/java/org/dspace/content/service/ItemService.java]: @pnbecker , I updated the javadocs for the new methods as well as the legacy methods to clarify which set return each group of items.

**Review by pnbecker — APPROVED** [review]
Thanks for clarifying @terrywbrady.
+1

File: dspace-rest/src/main/java/org/dspace/rest/common/FilteredCollection.java [hunk 2]
```diff
@@ -111,15 +111,15 @@ private void setup(org.dspace.content.Collection collection, ServletContext serv
         
         this.setNumberItemsProcessed(0);
         if (itemFilters.size() > 0) {
-        	Iterator<org.dspace.content.Item> childItems = itemService.findByCollection(context, collection, limit, offset);
+        	Iterator<org.dspace.content.Item> childItems = itemService.findAllByCollection(context, collection, limit, offset);
```
Comment (tdonohue) [inline dspace-rest/src/main/java/org/dspace/rest/common/FilteredCollection.java]: Just wanted to verify that this isn't a potential security issue, and that something is authorizing users to see all the items (including withdrawn ones) in the Collection.  Obviously, a change like this needs to be checked for security...we don't want to accidentally return withdrawn items in REST API requests, if you are not authorized to see them.

Comment (terrywbrady) [reply dspace-rest/src/main/java/org/dspace/rest/common/FilteredCollection.java]: @tdonohue , I ran the following query (1)unauthenticated and then (2)authenticated as admin in a collection with 1 withdrawn item.

/rest/filtered-collections/<coll-uuid>?limit=500&filters=is_withdrawn&expand=items

The withdrawn item count was 0 when unauthenticated, and no withdrawn items were returned.

When authenticated, a count of 1 was returned and the item was embedded in the result.

**Review by kshepherd — APPROVED** [review]
tested, works as advertised.
and i also see no withdrawn items unless i'm authenticated as an administrator.
thanks @terrywbrady ! will merge this now

**Pull request merged**